test/e2e/specs/editor/plugins/block-api.spec.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Using Block API', () => {
+	test( 'Inserts the filtered hello world block even when filter added after block registration', async ( {
+		page,
+		pageUtils,
+		requestUtils,
+	} ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-block-api' );
+		await pageUtils.createNewPost();
+
+		await pageUtils.insertBlock( { name: 'e2e-tests/hello-world' } );
+
+		const blockContent = await page.textContent(
+			'div[data-type="e2e-tests/hello-world"]'
+		);
+
+		expect( blockContent ).toBe( 'Hello Editor!' );
+
+		await requestUtils.deactivatePlugin( 'gutenberg-test-block-api' );
```

Comment:
Nitpick: Let's move this to `afterAll` so that when the test fails mid-way, it can still deactivate the plugin to do the cleanup.